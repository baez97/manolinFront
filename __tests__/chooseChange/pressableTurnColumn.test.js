import React from 'react';
import renderer from 'react-test-renderer';
import PressableTurnColumn from 
    '../../components/chooseChangeComponents/pressableTurnColumn';

jest.mock('../../components/dateUtils');
jest.unmock("react-native")
jest.unmock('react')
jest.unmock("react-test-renderer")
jest.unmock('expo');

// AUXILIAR FUNCTION TO MOCK A TURNS COLLECTION
function generateMockedTurns( day=25, month=2, year=2019, weekday=1 ) {
    const turnObj = {
        day   : day,
        month : month,
        year  : year,
        turn  : 'M',
        weekday: weekday
    };

    const turnObjsCollection = [];
    for ( let i = 0; i < 11; i++ ) {
        turnObjsCollection.push({...turnObj});
    }

    return turnObjsCollection;
}

describe('PressableTurnColumn', () => {

    describe("Methods", () => {
        it("handlePress calls to onPressFn of props", () => {
            const props = {
                onPressFn : jest.fn(),
                turnObjs  : ["FIRST_TURN", "SECOND_TURN"]
            };

            const pressCol = new PressableTurnColumn(props);
            pressCol.handlePress();

            expect( props.onPressFn ).toBeCalled();
            expect( props.onPressFn ).toBeCalledWith("FIRST_TURN");
        });
    });

    describe("View", () => {
        it("Matches the Snapshot when it is an ordinary day", () => {
            const turnCollection = generateMockedTurns();

            const rendered = renderer.create(
                <PressableTurnColumn turnObjs = { turnCollection } />
            );

            expect( rendered ).toMatchSnapshot();
        });

        it("Matches the Snapshot when it is a weekend day", () => {
            const turnCollection = generateMockedTurns(1, 1, 2019, 6);

            const rendered = renderer.create(<PressableTurnColumn turnObjs = { turnCollection } />);

            expect( rendered ).toMatchSnapshot();
        });

        it("Matches the Snapshot when it is today", () => {
            const turnCollection = generateMockedTurns(26, 2, 2019, 2);

            const rendered = renderer.create(<PressableTurnColumn turnObjs = { turnCollection } />);

            expect( rendered ).toMatchSnapshot();
        });
    });
});