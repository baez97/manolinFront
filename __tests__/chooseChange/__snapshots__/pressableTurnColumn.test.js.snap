// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PressableTurnColumn View Matches the Snapshot when it is a weekend day 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "width": 88.5,
      }
    }
  >
    <ExponentLinearGradient
      colors={
        Array [
          4279017174,
          4278481116,
        ]
      }
      endPoint={
        Array [
          0,
          1,
        ]
      }
      startPoint={
        Array [
          0,
          0,
        ]
      }
      style={
        Object {
          "height": 52.5,
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "white",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 35.416667660079604,
            "textAlign": "center",
          }
        }
      >
        1
      </Text>
    </ExponentLinearGradient>
    <ExponentLinearGradient
      colors={
        Array [
          4289914618,
          4285250006,
        ]
      }
      endPoint={
        Array [
          0,
          1,
        ]
      }
      startPoint={
        Array [
          0,
          0,
        ]
      }
      style={
        Object {
          "borderLeftColor": "#c5c9cc",
          "borderRightColor": "#c5c9cc",
          "borderRightWidth": 0.5,
          "paddingBottom": 17.5,
          "paddingLeft": 17.5,
          "paddingRight": 17.5,
          "paddingTop": 17.5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
    </ExponentLinearGradient>
  </View>
</View>
`;

exports[`PressableTurnColumn View Matches the Snapshot when it is an ordinary day 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "width": 88.5,
      }
    }
  >
    <ExponentLinearGradient
      colors={
        Array [
          4279017174,
          4278481116,
        ]
      }
      endPoint={
        Array [
          0,
          1,
        ]
      }
      startPoint={
        Array [
          0,
          0,
        ]
      }
      style={
        Object {
          "height": 52.5,
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "white",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 35.416667660079604,
            "textAlign": "center",
          }
        }
      >
        25
      </Text>
    </ExponentLinearGradient>
    <ExponentLinearGradient
      colors={
        Array [
          4294572537,
          4291742924,
        ]
      }
      endPoint={
        Array [
          0,
          1,
        ]
      }
      startPoint={
        Array [
          0,
          0,
        ]
      }
      style={
        Object {
          "borderLeftColor": "#c5c9cc",
          "borderRightColor": "#c5c9cc",
          "borderRightWidth": 0.5,
          "paddingBottom": 17.5,
          "paddingLeft": 17.5,
          "paddingRight": 17.5,
          "paddingTop": 17.5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "black",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
    </ExponentLinearGradient>
  </View>
</View>
`;

exports[`PressableTurnColumn View Matches the Snapshot when it is today 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "width": 88.5,
      }
    }
  >
    <ExponentLinearGradient
      colors={
        Array [
          4279017174,
          4278481116,
        ]
      }
      endPoint={
        Array [
          0,
          1,
        ]
      }
      startPoint={
        Array [
          0,
          0,
        ]
      }
      style={
        Object {
          "height": 52.5,
          "justifyContent": "center",
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "white",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 35.416667660079604,
            "textAlign": "center",
          }
        }
      >
        26
      </Text>
    </ExponentLinearGradient>
    <ExponentLinearGradient
      colors={
        Array [
          4294572537,
          4291742924,
        ]
      }
      endPoint={
        Array [
          0,
          1,
        ]
      }
      startPoint={
        Array [
          0,
          0,
        ]
      }
      style={
        Object {
          "borderLeftColor": "#c5c9cc",
          "borderRightColor": "#c5c9cc",
          "borderRightWidth": 0.5,
          "paddingBottom": 17.5,
          "paddingLeft": 17.5,
          "paddingRight": 17.5,
          "paddingTop": 17.5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "#056ec9",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "#056ec9",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "#056ec9",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "#056ec9",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "#056ec9",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "#056ec9",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "#056ec9",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "#056ec9",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "#056ec9",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "#056ec9",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
      <Text
        allowFontScaling={false}
        style={
          Object {
            "color": "#056ec9",
            "fontFamily": "montserrat-extra-bold",
            "fontSize": 40.7291678090925,
            "paddingBottom": 19.25,
            "textAlign": "center",
          }
        }
      >
        M
      </Text>
    </ExponentLinearGradient>
  </View>
</View>
`;
